classification1.Rmd

```diff
@@ -472,86 +489,155 @@ perim_concav + annotate("path",
 variables for two observations $a$ and $b$, i.e., 
 $a = (a_{1}, a_{2}, \dots, a_{m})$ and
 $b = (b_{1}, b_{2}, \dots, b_{m})$.
-Previously, when we had two variables, we added up the squared difference between each of our (two) variables,
-and then took the square root. Now we will do the same, except for *all* of our
-$m$ variables.  In other words, the distance formula becomes
+
+The distance formula becomes
 
 $$\mathrm{Distance} = \sqrt{(a_{1} -b_{1})^2 + (a_{2} - b_{2})^2 + \dots + (a_{m} - b_{m})^2}.$$
 
 This formula still corresponds to a straight-line distance, just in a space with 
-more dimensions. For example, suppose we decide to use 3 predictor variables (so, a 3-dimensional space):
-perimeter, concavity, and symmetry. Figure \@ref(fig:05-more) shows what
-the data look like when we visualize them as a 3-dimensional scatter.
+more dimensions. Suppose we want to calculate the distance between a new observation 
+that we want to classify with a perimeter of 0, concavity of 3.5 and symmetry of 1 and
+another observation with a perimeter, concavity and symmetry of 0.417, 2.31 and 0.837 respectively. 
+We have three predictor variables, perimeter, concavity, and symmetry (so, a 3-dimensional space) for two observations where
+$\mathrm{observation}_{old} = (0.417, 2.31, 0.837)$ and $\mathrm{observation}_{new} = (0, 3.5, 1)$.
+Previously, when we had two variables, we added up the squared difference between each of our (two) variables,
+and then took the square root. Now we will do the same, except for our
+three variables.  We calculate the distance as follows
+
+$$\mathrm{Distance} =\sqrt{(0 - 0.417)^2 + (3.5 - 2.31)^2 + (1 - 0.837)^2} = 1.27.$$
 In this case, the formula above is just the straight line distance in this 3-dimensional space.
+Let's calculate the distances between our new observation and each of the observations in the training set to find the $K=5$ neighbors when we have these three predictors. 
+```{r}
+new_obs_Perimeter <- 0
+new_obs_Concavity <- 3.5
+new_obs_Symmetry <- 1
 
+cancer |>
+  select(ID, Perimeter, Concavity, Symmetry, Class) |>
+  mutate(dist_from_new = sqrt((Perimeter - new_obs_Perimeter)^2 + 
+                              (Concavity - new_obs_Concavity)^2 +
+                                (Symmetry - new_obs_Symmetry)^2)) |>
+  arrange(dist_from_new) |>
+  slice(1:5) # subset the first 5 rows
+```
+Based on $K=5$ nearest neighbors with these three predictors we would classify the new observation as malignant since 4 out of 5 of the nearest neighbors are malignant class. 
+Figure \@ref(fig:05-more) shows what the data look like when we visualize them 
+as a 3-dimensional scatter with lines from the new observation to its five nearest neighbors.
 
 ```{r 05-more, echo = FALSE, message = FALSE, fig.cap = "3D scatter plot of the symmetry, concavity, and perimeter variables."}
 library(plotly)
-cancer |>
-  plot_ly(
-    x = ~Perimeter,
+attrs <- c("Perimeter", "Concavity", "Symmetry")
```

Comment:
this scatter is so incredibly cool. well done.

you should make an issue to remind us to somehow deal with this for the PDF output though (since we obviously can't do interactive 3D, so need to modify both the figure and the text after it)

Reply:
Made a new issue #241